src/main/java/com/snowflake/kafka/connector/internal/StageFilesProcessor.java

```diff
@@ -0,0 +1,622 @@
+package com.snowflake.kafka.connector.internal;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryPipeCreation;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryPipeStatus;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryService;
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.snowflake.ingest.connection.HistoryResponse;
+
+/**
+ * StageFileProcessor - class responsible for tracking files submitted to snowpipe for processing.
+ * This class is responsible for: - purging files with invalid offset - i.e. crash happened before
+ * offset was committed, but file was uploaded: it will be deleted from stage; - purging files which
+ * have been successfully ingested - i.e. their load status is LOADED - moving failed files (their
+ * status is FAILED or PARTIALLY_LOADED) or old files (1h) where no status is available, to the
+ * table stage.
```

Comment:
Thanks for javadocs but
minor nit:
can you use `<ul> <li>`, it will be more readable?